streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -104,6 +105,10 @@ private void putAndMaybeForward(final ThreadCache.DirtyEntry entry, final Intern
             } else {
                 underlying.put(entry.key(), entry.newValue());
             }
+        } catch (final ProcessorStateException e) {
```

Comment:
Thinking about this, I am wondering if caching stores are the best place to fix this. Caching might be disabled and thus, the fix would not cover this case?

Reply:
@mjsax , Just trying to understand. On the point `Caching might be disabled`, do you mean kafka wouldn't be supporting this it in future or User have config option to disable Caching?.

Reply:
@mjsax And does it applies to other Caching files too `CachingSessionStore.java` and `CachingWindowStore.java`?

Reply:
I refer to use configs. Users can disable caching. And yes it, also applies to the other caching stores. I think it would be better to use `MeteredXXX` stores -- those are always used as wrappers and thus fix the issues.

Reply:
Fixed